File: SwiatZwierzat_C++/Pike.cpp
#pragma once
#include "Fish.cpp"
class Pike : public Fish{ //*dziedziczenie* slowo public oznacza, że dziedziczone elementy (np. zmienne lub funkcje) mają taką widoczność jak w klasie bazowej, oprocz private w bazowej. Zapobiega to powielaniu tych samych cech obiektow. Dzieki temu mozemy
				//mozemy stworzyc oddzielna klase, czyli np Fish i odziedziczyc po niej zestaw cech które ma kazda ryba
public:
    Pike() : Fish(){ //wywolujemy konstruktor klasy, po ktorej dziedziczymy, bo szczupak tez jest ryba
        cout << "Jestem szczupakiem" << endl;
    }

    void swim() {
        cout << "Potrafie plywac" << endl;
    }

    void presentYourself() {
        cout << "Mam na imie " << getName() << " oraz zyje tu w wodzie juz od " << getAge() << " lat" << endl;
    }
};

File: SwiatZwierzat_Java/Pike.java
public class Pike extends Fish{ //*dziedziczenie* extends uzywamy podczas dziedziczenia klas. Zapobiega to powielaniu tych samych cech obiektow. Dzieki temu mozemy
    //możemy stworzyć oddzielną klasę, czyli Animal i odziedziczyć po niej zestaw cech które ma każde zwierzę

    Pike(){
        super(); //za pomoca slowa super w tym przypadku wywolujemy konstruktor klasy, po ktorej dziedziczymy
        System.out.println("Jestem szczupakiem");
    }

    @Override
    void swim() {
        System.out.println("Potrafie plywac");
    }

    @Override
    public void presentYourself() {
        System.out.println("Mam na imie " + getName() + " oraz zyje tu w wodzie juz od " + getAge() + " lat");
    }
}
